lib/tasks/database_cleanup.rake

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+namespace :database do
+  desc 'Check and remove orphaned objects'
+  task cleanup_orphans: :environment do
+    puts 'Checking and removing orphaned objects...'
+
+    # Tables to exclude from orphaned objects check
+    excluded_tables = ['accounts', 'audits', 'categories', 'category_types', 'cms_templates', 'legal_term_acceptances', 'legal_term_bindings',
+                      'legal_term_versions', 'proxy_logs', 'schema_migrations', 'service_cubert_infos', 'slugs', 'taggings', 'tags']
+
+    ids = Account.where(provider: true).pluck(:id)
+    # Iterate over tables with tenant_id field
+    tables = ActiveRecord::Base.connection.tables.select { |t| ActiveRecord::Base.connection.column_exists?(t, 'tenant_id') }
+
+    tables.each do |table|
```

Comment:
maybe it will be easier to go from Activerecord models to tables with `ApplicationRecord.descendants`. See `test/unit/models_test.rb`.